### PR TITLE
Add diagnostics

### PR DESCRIPTION
**Pull request by CopyDemon**
Fixes
Fix the UI panel layout, remove the variables panel, and change that panel to a multi-task panel. Now it can display diagnostics.

Proposed changes:
Create a diagnostics panel to display diagnostics.

Notice
Now the diagnostics only run on fixed JSON data, later when the API is up to date, will rewrite some functions in diagnostics.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by dangunter — COMMENTED** [review]
I assume this will be incorporated in a future PR that uses real/live diagnostic info. Should we move this PR back to draft?

**Comment by CopyDemon** [comment]
@dangunter you are right, I just turned this PR into a draft, and start to work on serving the new UI with FastAPI and do some mockups for diagnostics. Once the API part is cleaned up and UI is approved, I can come back and update the UI template in code.  if need some tests I still can use the old diagnostics UI template in this PR.

For now

**Comment by lbianchi-lbl** [comment]
The current plan is to merge #43 first and then this one (which is also expected to resolve the current CI failures).

File: idaes_ui/fv/fastAPI_route/api_post_diagnostics_runner.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            output_stream = io.StringIO()
+
+            # base on request.function_name run diagnostics function
+            match request.function_name:
+                case "display_underconstrained_set":
+                    dt.display_underconstrained_set(stream=output_stream)
+                case "display_constraints_with_large_residuals":
+                    dt.display_constraints_with_large_residuals(stream=output_stream)
+                case "display_potential_evaluation_errors":
+                    dt.display_potential_evaluation_errors(stream=output_stream)
```
Comment (lbianchi-lbl) [inline idaes_ui/fv/fastAPI_route/api_post_diagnostics_runner.py]:
```suggestion
            func = getattr(dt, request.function_name, None)
            if func is None:
                ...
                # TODO handle error
            func(stream=output_stream)
```

**Comment by dangunter** [comment]
@CopyDemon Could you rebase this on main? I see some merge conflicts there

**Comment by CopyDemon** [comment]
> Could you rebase this on main? I see some merge conflicts there

Working on it now

**Comment by dangunter** [comment]
@CopyDemon is this ready to review now?

**Comment by CopyDemon** [comment]
@dangunter This ticket is ready for review now:
1. Fixed the error that occurred when dragging the diagnostics panel.
2. Added a utility to show user notifications when refreshing, saving, and loading/refreshing diagnostics.
3. Added a utility to reset the layout.

**Comment by dangunter** [comment]
First question: why is the "dist" directory being checked into git? Aren't all those files generated by building the package?

File: IDAES-UI/dist/css/ag-grid.css
Comment (dangunter) [inline IDAES-UI/dist/css/ag-grid.css:1]: Remove dist/ directory from git

File: IDAES-UI/src/components/mosaic/mosic.tsx
Comment (dangunter) [inline IDAES-UI/src/components/mosaic/mosic.tsx:1]: This file needs to be (a) renamed: what does "mosaic" mean? We shouldn't be naming our code for the packages they are using; (b) commented: what does it do, why, who is calling it, what does it call, which are the major functions and what do they do?

**Comment by dangunter** [comment]
@CopyDemon I think we need to reduce the number of small files in deeply nested directories. See https://github.com/dangunter/idaes-ui/tree/add-diagnostics/IDAES-UI for a start on this, moving the entire tree under `src/components/header_components` to the top of `src` with 3 pairs of files
* Header.tsx, Header.module.css
* HeaderButtons.tsx, HeaderButtons.module.css
* DiagnosticsToggle.tsx, DiagnosticsToggle.module.css

This replaces 14 files with 4 levels of hierarchy (including `components/`) with 6 files and 1 level of hierarchy, yet retains flexibility and is easier to understand and maintain.

In the process, please use consistent case for all names in JavaScript and CSS, consistently use CSS modules (if we're going to use them at all, let's do it everywhere) and use names in the standard convention of "ComponentName.tsx" and "ComponentName.module.css". Also use your IDE to add spaces before curly braces, etc., in the standard styles for CSS and JS.

Note: the code in this tree is *not* tested and I can say with 99.99999% certainty it will not work -- I changed a bunch of names.

**Review by dangunter — CHANGES_REQUESTED** [review]
See prev comment

**Comment by dangunter** [comment]
From running the tool, there is another thing that needs to be fixed.
In the "Diagnostics Logs" tab of the lower panel, when you run a diagnostic function, it does not provide a horizontal scroll even when the text is wider than the screen. See screenshot below.

image

**Pull request merged**